### PR TITLE
Add scaffolding for the NPM 7 project

### LINKED ISSUE #4: Setup the repository with NPM 7+
# Description

The repository must be prepared to handle the new packages:
- #1 
- #2 
- #3 

## Requirements
- Setup NPM 7 with workspaces 
- Add a build script workflow for the build gate

### PR DESCRIPTION
**Pull request by janechu**
# Description

This code update includes:
- repository documentation
- github templates and workflows 
- codeowners file
- publishing configuration files
- linting configurations

This change uses the same format as the https://github.com/microsoft/fast-tooling repository, requiring NPM 7+ and NodeJs 16+.

## Issues

Closes #4 

## Reviewers

Note that this initial PR will fail as there is no workspaces to test against. This should be fine and expected as the next pull request will add a package.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{
```
Comment (EisenbergEffect) [inline package.json:1]: I think the root package.json for a monorepo should probably have `private` set to `true`.

**Review by EisenbergEffect — APPROVED** [review]
Pending one adjustment.

File: package.json (hunk 1 above)
Comment (janechu) [reply package.json:1]: It's on line 10

Comment (EisenbergEffect) [reply package.json:1]: Doh. Excellent!

**Review by williamw2 — APPROVED** [review]

**Pull request merged**